### DISCUSSION TITLE
CentOS 7 support

### DISCUSSION THREAD

**Discussion by cobrin** [starship/starship#3282] [Q&A]
Hi there,

Anxious to try out Starship

```
$ sh -c "$(curl -fsSL https://starship.rs/install.sh)" -- --platform unknown-linux-musl --bin-dir ~/bin`

$ eval "$(starship init bash)"
starship: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory

```

Are there any instructions to allow me to compile starship rather than take pre-existing binaries? 

Cheers

**Comment by chipbuster** [comment]
Whoa, that's...not supposed to happen. Something's changed in our musl builds---one of the reasons we made it was to avoid problems like that. I'll create an issue from this to track it.

In the meantime, if you're looking to compile starship, you'll need to install the latest version of rust. Get it from https://rustup.rs/. Then you can clone the repo, cd into it, and run `cargo build` and the build should complete.

Note that you'll probably need the development files for openssl to do this--I believe on CentOS this is under `openssl-devel`.

**Reply by chipbuster** [reply]
See https://github.com/starship/starship/issues/3286 for the tracking issue.
